File: 2208-minimum-operations-to-halve-array-sum/2208-minimum-operations-to-halve-array-sum.py
class Solution:
    def halveArray(self, nums: List[int]) -> int:
        
        totalSum = sum(nums)
        currentSum = totalSum
        totalSum /=2
        
        
        new_list = [-ele for ele in nums]
        
        heapify(new_list)
        
        ctr=0
        while currentSum > totalSum:
            
            #Make the new ele positive
            ele = - heappop(new_list) 
            
            ele /=2
            
            currentSum -= ele
            
            heappush(new_list, -ele)
            
            ctr+=1
            
            
        return ctr
            
        

File: 2208-minimum-operations-to-halve-array-sum/2208-minimum-operations-to-halve-array-sum.cpp
class Solution {
public:
    int halveArray(vector<int>& nums) {
        
        
        int N = nums.size();
        
        priority_queue<double> pq(nums.begin(), nums.end()); //max
        
        
        double sum = accumulate(nums.begin(), nums.end(), 0.0);
        
        double currentSum = sum;
        int ctr = 0;
        sum /= 2;
        
        while(currentSum > sum){
            
            
            double num = (double)pq.top()/2;
            pq.pop();
            
            currentSum -= num;
            
            pq.push(num);
            
            ctr++;
        } 
        
        return ctr;
    }
};